hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v1/CompletionTimeQueryViewV1.java

```diff
@@ -0,0 +1,239 @@
+        // ==============================================================
+        // LEGACY CODE
+        // ==============================================================
+        // Fixes the completion time to reflect the completion sequence correctly
+        // if the file slice base instant time is not in datetime format.
+        // For example, many test cases just use integer string as the instant time.
+        // CAUTION: this fix only works for OCC(Optimistic Concurrency Control).
+        // for NB-CC(Non-blocking Concurrency Control), the file slicing may be incorrect.
+        return Option.of(instantTime);
+      }
+    }
+    return completionTimeOpt;
+  }
+
+  @Override
+  public Option<String> getCompletionTime(String beginTime) {
+    String completionTime = this.beginToCompletionInstantTimeMap.get(beginTime);
+    if (completionTime != null) {
+      return Option.of(completionTime);
+    }
+
+    // ***This is the key change between V1 and V2 completion time query-view***
+    if (isArchived(beginTime)) {
+      // Completion time and begin time are same for archived instants.
+      return Option.of(beginTime);
+    }
+    // the instant is still pending
+    return Option.empty();
+  }
+
+  @Override
+  public List<String> getInstantTimes(
+      HoodieTimeline timeline,
+      Option<String> rangeStart,
+      Option<String> rangeEnd,
+      InstantRange.RangeType rangeType) {
+    // assumes any instant/transaction lasts at most 1 day to optimize the query efficiency.
+    return getInstantTimes(timeline, rangeStart, rangeEnd, rangeType, s -> HoodieInstantTimeGenerator.instantTimeMinusMillis(s, MILLI_SECONDS_IN_ONE_DAY));
+  }
+
+  @Override
+  @VisibleForTesting
+  public List<String> getInstantTimes(
+      String rangeStart,
+      String rangeEnd,
+      Function<String, String> earliestInstantTimeFunc) {
+    return getInstantTimes(metaClient.getCommitsTimeline().filterCompletedInstants(), Option.ofNullable(rangeStart), Option.ofNullable(rangeEnd),
+        InstantRange.RangeType.CLOSED_CLOSED, earliestInstantTimeFunc);
+  }
+
+  /**
+   * Queries the instant start time with given completion time range.
+   *
+   * @param timeline            The timeline.
+   * @param rangeStart              The query range start completion time.
+   * @param rangeEnd                The query range end completion time.
+   * @param rangeType               The range type.
+   * @param earliestInstantTimeFunc The function to generate the earliest start time boundary
+   *                                with the minimum completion time.
+   *
+   * @return The sorted instant time list.
+   */
+  private List<String> getInstantTimes(
+      HoodieTimeline timeline,
+      Option<String> rangeStart,
+      Option<String> rangeEnd,
+      InstantRange.RangeType rangeType,
+      Function<String, String> earliestInstantTimeFunc) {
+    throw new RuntimeException("Not Implemented Yet !");
```

Comment:
Incremental query view for V1 not yet implemented

Reply:
better error message here? `Incremental query view for timeline version 1 not yet implemented`